### PR TITLE
feat: introduce fastjson in converter

### PR DESCRIPTION
**Pull request by Yisaer**

### CODE REVIEW

File: internal/converter/json/converter.go [hunk 1]
```diff
@@ -40,3 +44,280 @@ func (c *Converter) Decode(b []byte) (interface{}, error) {
...
+}
+
+func NewFastJsonConverter(schema map[string]*ast.JsonStreamField) *FastJsonConverter {
+	return &FastJsonConverter{
+		schema: schema,
+	}
+}
+
+func (c *FastJsonConverter) Encode(_ interface{}) ([]byte, error) {
+	panic("FastJsonConverter shouldn't covert json data")
```
Comment (ngjaying) [inline internal/converter/json/converter.go:59]: what's the plan for this?

File: internal/topo/planner/planner.go [hunk 2]
```diff
@@ -194,7 +194,12 @@ func transformSourceNode(t *DataSourcePlan, sources []*node.SourceNode, options
 		}
 		var srcNode *node.SourceNode
 		if len(sources) == 0 {
-			sourceNode := node.NewSourceNode(string(t.name), t.streamStmt.StreamType, pp, t.streamStmt.Options, options.SendError)
+			var sourceNode *node.SourceNode
+			if !t.isSchemaless {
+				sourceNode = node.NewSourceNode(string(t.name), t.streamStmt.StreamType, pp, t.streamStmt.Options, options.SendError, t.streamFields)
+			} else {
+				sourceNode = node.NewSourceNode(string(t.name), t.streamStmt.StreamType, pp, t.streamStmt.Options, options.SendError, nil)
```
Comment (ngjaying) [inline internal/topo/planner/planner.go:201]: Is t.streamFields == nil in else? If so, can we just eliminate the if/else

File: internal/topo/planner/planner.go [hunk 3]
```diff
@@ -213,7 +218,11 @@ func transformSourceNode(t *DataSourcePlan, sources []*node.SourceNode, options
 			srcNode = getMockSource(sources, string(t.name))
 		}
 		if srcNode == nil {
-			srcNode = node.NewSourceNode(string(t.name), t.streamStmt.StreamType, pp, t.streamStmt.Options, options.SendError)
+			if t.isSchemaless {
```
Comment (ngjaying) [inline internal/topo/planner/planner.go:221]: same as the before comment

**Review by ngjaying — COMMENTED** [review]
some comments

File: internal/topo/planner/planner.go [hunk 4]
```diff
@@ -194,7 +194,12 @@ func transformSourceNode(t *DataSourcePlan, sources []*node.SourceNode, options
 		}
 		var srcNode *node.SourceNode
 		if len(sources) == 0 {
-			sourceNode := node.NewSourceNode(string(t.name), t.streamStmt.StreamType, pp, t.streamStmt.Options, options.SendError)
+			var sourceNode *node.SourceNode
+			schema := t.streamFields
+			if t.isSchemaless {
+				schema = nil
+			}
```
Comment (Yisaer) [inline internal/topo/planner/planner.go:201]: It seems we should clear the schema in order to pass the test

**Review by ngjaying — APPROVED** [review]

**Pull request merged**